libexec/test-server.ts

```diff
@@ -2,14 +2,19 @@
  * @file run this file if you want to start test server and test yourself
  */
 
-import {spawn} from "child_process";
+import {ChildProcess, spawn} from "child_process";
+import {existsSync} from "fs";
 
 
-export function startServer(onClose: (...args) => unknown):void {
+export function startServer(onClose: (...args) => unknown): ChildProcess {
+  if (!existsSync('dist/test.html')) {
+    throw new Error("dist/test.html not found; do you need to run a build first?")
+  }
 
-  spawn('npx', ['http-server', 'dist'], {shell: true})
-  const waitOnProcess = spawn('npx', ['wait-on', 'http-get://localhost:8080/test.html'], {shell: true})
+  const process = spawn('npx', ['ws', '-p', '8080', '-d', 'dist'], {stdio: "inherit"})
```

Comment:
Can you explain why we're starting separate processes for both the server and Cypress? Just for my own edification.

Reply:
You are 100% correct that this is not a great way of doing things. The http server, the liveness check, and cypress itself can all together in the same node process happily, and with no chance of accidentally leaving an HTTP server running in the background.

That said, I’m just doing the minimum here to be able to write a single test that itwêwina changes don’t break click-in-text, and to avoid having any known-insecure packages in code to be packaged into a browser extension and then loaded automatically into all sorts of webpages.

Reply:
Ah okay got it! I wasn't gonna suggest a change because I assumed there was a good reason for spawning the process I wasn't aware of, but that's good to know. Thanks for indulging my curiosity!